test/quantization/core/experimental/test_floatx.py

```diff
@@ -392,7 +392,37 @@ def test_save_load(self, dtype, device):
             torch.testing.assert_close(x1, x1_save_load, atol=0, rtol=0)
 
 
+class TestFloat4Dtype(TestCase):
+    # TODO(#146647): make the testing generic for shell dtypes
+    def test_float4_e2m1fn_x2(self, device):
+        # can create a tensor of dtype float4
+        x1 = torch.empty(4096, 4096, device=device, dtype=torch.float4_e2m1fn_x2)
```

Comment:
Question: How is the shape defined for this type? For a tensor of shape (4096, 4096), does it actually contain 4096 * 4096 *2 float4 values?

Reply:
I am trying to understand when I unpack, what would be the resulting shape of this tensor

Reply:
So this will construct a tensor of 4096 x 4096 elements and these elements are what we are calling a shell dtype. I think Vasily has a fuller description here: https://github.com/pytorch/pytorch/issues/146414

In this case case, the shell dtype is takes up 1 byte per element and semantically represents to fp4 values packed in to 1 byte.  And in its current states its really a "named" uint8 tensor that has meaning but very limited op coverage. 

TLDR:
Yeah it semantically represents 4096 * 4096 *2 values fp4 values

Reply:
yeah the former